packages/adminrouter/extra/src/test-harness/modules/generic_test_code/generalised_tests.py

```diff
@@ -427,11 +476,16 @@ def test_if_request_is_sent_to_correct_upstream(
             master_ar_process_perclass,
             valid_user_header,
             path,
-            expected_upstream):
+            expected_upstream,
+            vhost):
+
+        req_headers = copy.deepcopy(valid_user_header)
+        if vhost is not None:
+            req_headers = {"Host": vhost}
```

Comment:
This line overrides the deep copy above. Is that the intent? If so, this code is easier to read.
```python
if vhost:
  req_headers = {"Host": vhost"}
else:
  req_headers = copy.deepcopy(valid_user_header)
```

Reply:
It is a bug, the last line should be: `req_headers['Host'] = vhost`. Thanks!

Reply:
I think you missed this one. It should be the following, right?
```python
req_headers = copy.deepcopy(valid_user_header)
if vhost is not None:
     req_headers['Host'] = vhost
```